crates/taplo-cli/src/lib.rs

```diff
@@ -111,33 +120,67 @@ impl<E: Environment> Taplo<E> {
             };
         };
 
-        let patterns = patterns
-            .into_iter()
-            .unique()
-            .map(|p| glob::Pattern::new(&p).map(|_| p))
-            .collect::<Result<Vec<_>, _>>()?;
-
-        let files = patterns
-            .into_iter()
-            .map(|pat| self.env.glob_files_normalized(&pat))
-            .collect::<Result<Vec<_>, _>>()
-            .into_iter()
-            .flatten()
-            .flatten()
-            .collect::<Vec<_>>();
-
-        let total = files.len();
+        tracing::trace!("Compiled patterns");
 
-        let files = files
-            .into_iter()
-            .filter(|path| config.is_included(path))
-            .collect::<Vec<_>>();
+        let mut skip_patterns = vec![];
+        let mut keep_patterns = vec![];
 
-        let excluded = total - files.len();
-
-        tracing::info!(total, excluded, "found files");
+        let opts = glob::MatchOptions {
+            case_sensitive: true,
+            ..Default::default()
+        };
 
-        Ok(files)
+        let cwd = cwd.to_path_buf();
+
+        let mut bldr = ignore::WalkBuilder::new(&cwd);
+        bldr.git_ignore(true)
+            .git_exclude(true)
+            .git_global(true)
+            .ignore(true)
+            .hidden(false)
+            .same_file_system(true);
```

Comment:
I feel the same as you that obeying gitignore rules is likely a desired behavior and i personally have nothing against this becoming the default behavior (one day)

however, I agree with @panekj https://github.com/tamasfe/taplo/issues/619#issuecomment-2162838457 that this can be viewed as  a breaking change, so would be better to have this as an opt-in feature, at least for now.

Also, given how `ignore` API is designed -- _some ignore rules by default_ --  the walk builder should be configured with `standard_filters(false)` first, then enable what we want.

Question
is switching to `ignore` create sufficient to prune large target directly without recursing to it?